File: backend/src/main/java/dev/cristianinbits/hyron/hyrox/service/HyroxService.java
package dev.cristianinbits.hyron.hyrox.service;

import dev.cristianinbits.hyron.hyrox.dto.HyroxDetailsCreateRequest;
import dev.cristianinbits.hyron.hyrox.dto.HyroxDetailsResponse;

public interface HyroxService {
    HyroxDetailsResponse createOrUpdateDetails(Long userId, Long workoutId, HyroxDetailsCreateRequest request);

    HyroxDetailsResponse getDetails(Long userId, Long workoutId);

    void deleteDetails(Long userId, Long workoutId);
}

File: frontend/src/services/hyroxService.ts
import api from './api'
import type { HyroxDetailsResponse, HyroxDetailsCreateRequest } from '../types/hyrox'

export const hyroxService = {
    getDetails: async (userId: number, workoutId: number): Promise<HyroxDetailsResponse> => {
        const response = await api.get(`/users/${userId}/workouts/${workoutId}/hyrox`)
        return response.data
    },

    saveDetails: async (userId: number, workoutId: number, data: HyroxDetailsCreateRequest): Promise<HyroxDetailsResponse> => {
        const response = await api.put(`/users/${userId}/workouts/${workoutId}/hyrox`, data)
        return response.data
    },

    deleteDetails: async (userId: number, workoutId: number): Promise<void> => {
        await api.delete(`/users/${userId}/workouts/${workoutId}/hyrox`)
    },
}